python/packages/core/agent_framework/_workflows/_agent_executor.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from dataclasses import dataclass
+from typing import Any
+
+from .._agents import AgentProtocol
+from .._threads import AgentThread
+from .._types import AgentRunResponse, AgentRunResponseUpdate, ChatMessage
+from ._events import (
+    AgentRunEvent,
+    AgentRunUpdateEvent,  # type: ignore[reportPrivateUsage]
+)
+from ._executor import Executor, handler
+from ._workflow_context import WorkflowContext
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
```

Comment:
Ideally we also don't use data classes as there might be bugs and/or slight differences between python version with them, so plain vanilla classes are the way (and we do still have to apply this across the codebase)

Reply:
Okay

Reply:
I think we should make this change in future PRs. Several places in serialization/deserialization likely assume message is some data class type.